src/Approximations/distance.jl

```diff
@@ -50,3 +50,51 @@ function distance(H1::AbstractHyperrectangle{N},
     end
     return norm(d, p)
 end
+
+"""
+    distance(x::AbstractVector, y::AbstractVector, p::Real=2.0)
+
+Compute the distance between two vectors with respect to the given `p`-norm,
+computed as
+
+```math
+    \\|x - y\\|_p = \\left( \\sum_{i=1}^n | x_i - y_i |^p \\right)^{1/p}
+```
+
+### Input
+
+- `x` -- vector
+- `y` -- vector
+- `p` -- (optional, default: `2.0`) the `p`-norm used; `p = 2.0` corresponds to
+         the usual Euclidean norm
+
+### Output
+
+A scalar representing ``‖ x - y ‖_p``.
+"""
+function distance(x::AbstractVector, y::AbstractVector, p::Real=2.0)
+    return norm(x - y, p)
+end
+
+"""
+    distance(x::AbstractVector, L::Line, p::Real=2.0)
+
+Compute the distance between point `x` and the line with respect to the given
+`p`-norm.
+
+### Input
+
+- `x` -- vector
+- `L` -- line
+- `p` -- (optional, default: `2.0`) the `p`-norm used; `p = 2.0` corresponds to
+         the usual Euclidean norm
+
+### Output
+
+A scalar representing the distance between `x` and the line `L`.
+"""
+function distance(x::AbstractVector, L::Line, p::Real=2.0)
+    n = L.n  # direction of the line
+    t = dot(x - L.p, n) / dot(n, n)
+    return distance(x, L.p + t*n, p)
+end
```

Comment:
i stored all `distance` functions in this file, because it's called `distance.jl` and contains `distance(H1::AbstractHyperrectangle{N}, H2::AbstractHyperrectangle{N}; p::Real=2)`. I can refactor this method to `Line.jl` if you prefer.

Reply:
My thoughts are that this function might be helpful in `LazySets` and `Approximations` is not a dependency.